llvm/lib/CodeGen/AsmPrinter/CodeViewDebug.cpp

```diff
@@ -250,7 +250,10 @@ CodeViewDebug::getInlineSite(const DILocation *InlinedAt,
         InlinedAt->getLine(), InlinedAt->getColumn(), SMLoc());
     Site->Inlinee = Inlinee;
     InlinedSubprograms.insert(Inlinee);
-    getFuncIdForSubprogram(Inlinee);
+    auto InlineeIdx = getFuncIdForSubprogram(Inlinee);
+
+    if (ParentFuncId == CurFn->FuncId)
```

Comment:
I think the condition `InlinedAt->getInlinedAt() == nullptr` would more clearly correspond to directly inlined children. Or, you could reuse `OuterIA` with more changes.